BitmarkRegistry/Models/AssetR.swift

```diff
@@ -37,8 +37,44 @@ class AssetR: Object {
     self.status = asset.status
     self.offset = asset.offset
     self.createdAt = asset.created_at
+
+    if let currentAccount = Global.currentAccount {
+      self.assetFilePath = try? AssetFileService(owner: currentAccount, assetId: asset.id).getAssetFile()?.lastPathComponent
+    }
+
     asset.metadata.forEach { (key, value) in
       metadata.append(MetadataR(value: [key, value]))
     }
   }
+
+  // MARK: - Data Handlers
+  func updateAssetFilePath(_ assetFilePath: String, completion: @escaping () -> Void) {
+    guard let currentAccount = Global.currentAccount else { return }
+    do {
+      let userConfiguration = try RealmConfig.user(currentAccount.getAccountNumber()).configuration()
+      let userRealm = try Realm(configuration: userConfiguration)
+
+      try userRealm.write {
+        self.assetFilePath = assetFilePath.lastPathComponent
+      }
+
+      completion()
+    } catch {
+      ErrorReporting.report(error: error)
+    }
+  }
+
+  func getAssetType() -> AssetType {
```

Comment:
Realm model is purely to represent data and interact with RealmDB.
Doing other business logics here is not a good idea.

Reply:
I'm not really sure about this.
to my think, this is as the model - object; which is in interact with RealmDB.
and in this case, assetR object is having a specific assetType property 🤔